voice/src/main/kotlin/FrameInterceptor.kt

```diff
@@ -0,0 +1,86 @@
+package dev.kord.voice
+
+import dev.kord.common.annotation.KordVoice
+import dev.kord.gateway.Gateway
+import dev.kord.voice.gateway.Speaking
+import dev.kord.voice.gateway.VoiceGateway
+import kotlin.properties.Delegates
+
+/**
+ * Variables that are accessible to any FrameInterceptor through the [VoiceConnection.frameInterceptorFactory].
+ *
+ * @param gateway the gateway that handles the guild this voice connection is connected to.
+ * @param voiceGateway the underlying [VoiceGateway].
+ * @param ssrc the current SSRC retrieved from Discord.
+ */
+@KordVoice
+data class FrameInterceptorContext(
+    val gateway: Gateway,
+    val voiceGateway: VoiceGateway,
+    val ssrc: Int,
+)
+
+@KordVoice
+internal class FrameInterceptorContextBuilder {
+    var gateway: Gateway by Delegates.notNull()
+    var voiceGateway: VoiceGateway by Delegates.notNull()
```

Comment:
provide those in the constructor of the class if they are mandatory while still allowing them to be edited inside the builder using `var`